### PR TITLE
cmd | node: Refactor core flags

### PR DESCRIPTION
**Pull request by renaynay**
## Description
This PR consolidates flags related to adding a core endpoint. The reasoning behind this is to prevent users from having to pass too many flags once #571 is resolved (since the core accessor will now need both a `gRPC` and `RPC` endpoint for the celestia-core connection).

## New flags

**In the case that the default ports are exposed (grpc:9090, rpc:26657) on the celestia-core endpoint**:
```
celestia <node_type> start --core <ip addr of core endpoint>
```
**In the case that custom ports are exposed on the celestia-core endpoint**:
```
celestia <node_type> start --core <ip addr of core endpoint> --core.grpc 9090 --core.rpc 26657
```

If `--core.grpc` || `--core.rpc` are not provided, we assume the defaults and attempt to connect.

### CODE REVIEW

File: core/client.go [hunk 1]
```diff
@@ -13,14 +13,14 @@ import (
 type Client = client.Client
 
 // NewRemote creates a new Client that communicates with a remote Core endpoint over HTTP.
-func NewRemote(protocol, remoteAddr string) (Client, error) {
+func NewRemote(endpoint string) (Client, error) {
```
Comment (Wondertan) [inline core/client.go:16]: I would make this constructor receive a port with an IP as a separate params, as you do now in CoreAccessor.

**Comment by renaynay** [comment]
from @YazzyYaz 

Immediate thoughts are, it should be explicit:
so --core should mean it’s for the IP address so something like --core.ip
or .grpc and .rpc  I’d rather have something like --core.grpc.port and --core.rpc.port since it relates to ports.
my 2 cents (edited)

File: cmd/flags_core.go [hunk 2]
```diff
@@ -12,70 +12,76 @@ import (
...
+		_, err = strconv.Atoi(grpc)
 		if err != nil {
 			return err
 		}
-		env.AddOptions(node.WithGRPCEndpoint(grpcURL.Host))
+		env.AddOptions(node.WithGRPCPort(grpc))
 	}
-
+	// TODO @renaynay: implement some sort of sanity check to make sure grpc/rpc aren't provided without
+	//  --core also being specified?
```
[the remark this answers is not in the document]
Comment (renaynay) [reply cmd/flags_core.go]: https://github.com/celestiaorg/celestia-node/pull/910/commits/164b0b9e912fe7cf6316b091650927fb61655087

**Comment by renaynay** [comment]
@YazzyYaz https://github.com/celestiaorg/celestia-node/pull/910/commits/51925630b5186f43a212c6d1e96d9acdf68591f8

File: node/config_opts.go [hunk 3]
```diff
@@ -1,18 +1,18 @@
 package node
 
 // WithRemoteCore configures Node to start with remote Core.
-func WithRemoteCore(protocol string, address string) Option {
+func WithRemoteCore(ip, port string) Option {
```
[the remark this answers is not in the document]
Comment (renaynay) [reply node/config_opts.go]: https://github.com/celestiaorg/celestia-node/pull/910/commits/fffeaa00cacc5f4321d72f7a94f4dc2b0dfaa189

File: core/client.go (hunk 1 above)
Comment (renaynay) [reply core/client.go]: https://github.com/celestiaorg/celestia-node/pull/910/commits/5889004444a76ba75580b645cc785cadb69cac71

File: cmd/flags_core.go [hunk 4]
```diff
@@ -12,70 +12,78 @@ import (
...
+		_, err = strconv.Atoi(grpc)
 		if err != nil {
 			return err
 		}
-		env.AddOptions(node.WithGRPCEndpoint(grpcURL.Host))
+		env.AddOptions(node.WithGRPCPort(grpc))
+		return nil
+	}
+	if cmd.Flag(coreGRPCFlag).Changed || cmd.Flag(coreRPCFlag).Changed {
+		return fmt.Errorf("cannot specify RPC/gRPC ports without specifying an IP address for --core.ip")
```
Comment (Wondertan) [inline cmd/flags_core.go:72]: Nice. The only problem here is that any new added core flag parsing logic would need to be nested in the if above or coreIP plag to be always In the end, which is fine

**Review by Wondertan — APPROVED** [review]
LGTM, only nits

File: cmd/flags_core.go (hunk 4 above)
Comment (renaynay) [reply cmd/flags_core.go]: https://github.com/celestiaorg/celestia-node/pull/910/commits/8ccb595089e1d071c1c0bd9a40ddce1c2d7bd769

File: node/config_opts.go [hunk 5]
```diff
@@ -1,18 +1,24 @@
...
 	return func(sets *settings) {
-		sets.cfg.Core.Protocol = protocol
-		sets.cfg.Core.RemoteAddr = address
+		sets.cfg.Core.IP = ip
 	}
 }
 
-// WithGRPCEndpoint configures Node to connect to given gRPC address
+// WithRemoteCoreRPCPort configures Node to connect to the given remote Core port.
+func WithRemoteCoreRPCPort(port string) Option {
```
[the remark this answers is not in the document]
Comment (renaynay) [reply node/config_opts.go]: https://github.com/celestiaorg/celestia-node/pull/910/commits/8ccb595089e1d071c1c0bd9a40ddce1c2d7bd769

File: cmd/flags_core.go [hunk 6]
```diff
@@ -12,70 +12,78 @@ import (
...
+	return nil
+}
 
-	if _, _, err := net.SplitHostPort(u.Host); err != nil {
-		return "", "", fmt.Errorf("incorrect address provided for Remote Core")
+// sanityCheckIP trims leading protocol scheme and port from the given
+// IP address if present.
+func sanityCheckIP(ip string) (string, error) {
+	original := ip
+	ip = strings.TrimPrefix(ip, "http://")
```
Comment (adlerjohn) [inline cmd/flags_core.go:81]: A hit hesitant on the manual parsing of IP format, but I guess them's the breaks

**Review by adlerjohn — APPROVED** [review]
utACK

File: cmd/flags_core.go (hunk 6 above)
Comment (adlerjohn) [reply cmd/flags_core.go:81]: (feel free to resolve conversation at will)

**Pull request merged**